src/main/java/uk/gov/companieshouse/api/strikeoffobjections/service/impl/CompanyProfileService.java

```diff
@@ -1,20 +1,60 @@
 package uk.gov.companieshouse.api.strikeoffobjections.service.impl;
 
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.ApiClient;
+import uk.gov.companieshouse.api.handler.exception.URIValidationException;
 import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.api.strikeoffobjections.common.ApiLogger;
+import uk.gov.companieshouse.api.strikeoffobjections.common.LogConstants;
 import uk.gov.companieshouse.api.strikeoffobjections.service.ICompanyProfileService;
+import uk.gov.companieshouse.service.ServiceException;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
 
 @Service
 public class CompanyProfileService implements ICompanyProfileService {
-    @Override
-    public CompanyProfileApi getCompanyProfile(String requestId, String companyNumber) {
+    private static final String LOG_COMPANY_NUMBER_KEY = LogConstants.COMPANY_NUMBER.getValue();
+
+    private ApiSdkClient apiSdkClient;
+    private ApiLogger apiLogger;
+
+    @Autowired
+    public CompanyProfileService(ApiSdkClient apiSdkClient,
+                                 ApiLogger apiLogger) {
+        this.apiSdkClient = apiSdkClient;
+        this.apiLogger = apiLogger;
+    }
+
+    /**
+     * Calls the API SDK Manager Java library in order to retrieve the company (profile) details.
+     */
+    public CompanyProfileApi getCompanyProfile(String requestId, String companyNumber) throws ServiceException {
```

Comment:
it's only a tiny thing, I've started to put the params in order of importance, might be better to have company number before requestId?